docs/upgrade.md

```diff
@@ -0,0 +1,245 @@
+
+```sh
+npx storybook@next migrate csf-2-to-3 --glob="src/**/*.stories.js"
+```
+
+### storiesOf to CSF
+
+Storybook 7's architecture is focused on performance and needs code that is statically analyzable. For that reason, it does not work with `storiesOf`. We provide a codemod which, in most cases, should automatically make the code changes for you (make sure to update the glob to fit your files):
+
+```sh
+npx storybook@next migrate storiesof-to-csf --glob="src/**/*.stories.tsx"
+```
+
+### .stories.mdx to MDX+CSF
+
+Storybook 7 provides a cleaner [docs](./writing-docs/introduction.md) that defines manual documentation in pure MDX and stories in CSF, rather than the previous `.stories.mdx` hybrid approach, which is now deprecated. You can automatically convert your files using the following codemod (make sure to update the glob to fit your files):
+
+```sh
+npx storybook@next migrate mdx-to-csf --glob "src/**/*.stories.mdx"
+```
+
+You’ll also need to update your `stories` glob in `.storybook/main.js` to include the newly created `.mdx` and `.stories.js` files if it doesn’t already.
+
+**NOTE:** this migration supports the Storybook 6 [“CSF stories with MDX docs”](https://github.com/storybookjs/storybook/blob/6e19f0fe426d58f0f7981a42c3d0b0384fab49b1/code/addons/docs/docs/recipes.md#csf-stories-with-mdx-docs) recipe.
+
+### Storybook 6 recipe “CSF stories with arbitrary MDX”
+
+Storybook 6 had a second recipe, [“CSF stories with arbitrary MDX”](https://github.com/storybookjs/storybook/blob/6e19f0fe426d58f0f7981a42c3d0b0384fab49b1/code/addons/docs/docs/recipes.md#csf-stories-with-arbitrary-mdx). This recipe is still supported in Storybook 7, but the new recommendation is cleaner & more future proof. Upgrade manually using the following example:
+
+#### Before
+
+```mdx
+// Button.mdx
+
+import { Story } from '@storybook/addon-docs';
+
+# Story embed
+
+<Story id="button--basic" />
+```
+
+```js
+// Button.stories.js
+
+import { Button } from './Button';
+import mdx from './Button.mdx';
+
+export default {
+  parameters: { docs: { page: mdx } },
+  component: Button,
+};
+```
+
+```js
+// .storybook/main.js
+
+module.exports = {
+  stories: ['../src/**.stories.@(js|jsx|ts|tsx)'],
+};
+```
+
+#### After
+
+Your MDX files should be updated to reference the CSF story export, which is safer and cleaner than referencing the string ID:
+
+<!-- prettier-ignore-start -->
+```mdx
+{/* Button.mdx */}
+
+{/* Note the new package we import from */}
+import { Story } from '@storybook/blocks';
+import * as ButtonStories from './Button.stories';
+
+# Story embed
+
+<Story of={ButtonStories.Basic} />
+```
+<!-- prettier-ignore-end -->
+
+You should also remove the MDX imports from your story files:
+
+```js
+// Button.stories.js
+
+import { Button } from './Button';
+
+export default {
+  component: Button,
+};
+```
+
+Finally, you’ll to add the MDX files to `.storybook/main.js` so that they are indexed:
```

Comment:
Shouldn't this be "Finally, you'll need to add the MDX files....." or something in that nature?